packages/core/test/render3/directive_spec.ts

```diff
@@ -121,6 +122,104 @@ describe('directive', () => {
          expect(directiveInstance !.testValue).toBe(false);
        });
 
+    it('should match directives on <ng-template>', () => {
+      /**
+       *   @Directive({
+       *     selector: 'ng-template[directiveA]'
+       *   })
+       *   export class DirectiveA {
+       *     constructor(public templateRef: TemplateRef<any>) {}
+       *   }
+       */
+      let tmplRef: any;
+      class DirectiveA {
+        constructor(public templateRef: any) { tmplRef = templateRef; }
+        static ngDirectiveDef = defineDirective({
+          type: DirectiveA,
+          selectors: [['ng-template', 'directiveA', '']],
+          factory: () => new DirectiveA(directiveInject(TemplateRef as any))
+        });
+      }
+
+      function MyComponent_ng_template_Template_0(rf: RenderFlags, ctx: any) {
+        if (rf & 1) {
+          text(0, 'Some content');
+        }
+      }
+      class MyComponent {
+        static ngComponentDef = defineComponent({
+          type: MyComponent,
+          selectors: [['my-component']],
+          factory: () => new MyComponent(),
+          consts: 2,
```

Comment:
```suggestion
          consts: 1,
```

(...we're not generating 2 here, are we?)